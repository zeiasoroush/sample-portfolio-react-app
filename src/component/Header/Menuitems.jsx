import React, { Component } from 'react';
import { BrowserRouter as Router, Link } from 'react-router-dom';

class Menuitems extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isactiveId: 1,
      location: 'http://localhost:3000/'
    };
    this.isActive = this.isActive.bind(this);
    this.handleClick = this.handleClick.bind(this);
  }
  isActive(id, index) {
    if (this.state.isactiveId === 0 && index === 0) {
      return 'ui-tabs-active';
    } else {
      return id === this.state.isactiveId ? 'ui-tabs-active' : '';
    }
  }
  handleClick(e, id) {
    console.log(e);
    const event = e.currentTargethref;
    this.setState({ isactiveId: id, location: event }, () => {
      this.props.handlecallbackchild(this.state.location);
    });
  }
  render() {
    return (
      <ul>
        {this.props.listcontent.map((content, index) => {
          return (
            <li key={index}>
              <Router>
                <Link
                  className={this.isActive(content.id, index)}
                  onClick={e => this.handleClick(e, content.id)}
                  to={content.link}
                >
                  <img
                    src={require(`../../assets/images/icon-${content.id}.png`)}
                    alt={content.name}
                  />
                  <em>{content.name}</em>
                </Link>
              </Router>
            </li>
          );
        })}
      </ul>
    );
  }
}
export default Menuitems;
