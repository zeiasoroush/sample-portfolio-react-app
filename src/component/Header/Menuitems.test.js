import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Menuitems from './Menuitems';

describe('<menuitem />', () => {
  let mockfn = jest.fn();
  it('it render Header', () => {
    const childcomponent = shallow(
      <Menuitems listcontent={[{ id: 1, link: '/', name: 'HOMEPAGE' }]} />
    );
    expect(childcomponent).toMatchSnapshot();
  });
  it('specify active tab in header onclick event ', () => {
    Menuitems.prototype.handleClick = mockfn;
    const wrapper = shallow(
      <Menuitems
        listcontent={[{ id: 2, link: '/AboutUs', name: 'ABOUT US' }]}
        onclick={mockfn}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('"handleClick" to have been called ', () => {
    const props = {
      listItem: [
        {
          id: 1,
          link: '/',
          name: 'HOMEPAGE'
        },
        {
          id: 2,
          link: '/AboutUs',
          name: 'ABOUT US'
        },
        {
          id: 3,
          link: '/Gallery',
          name: 'OUR GALLERY'
        },
        {
          id: 4,
          link: '/ContactUs',
          name: 'CONTACT US'
        }
      ]
    };
    const handleClick = jest.fn();
    let wrapper = shallow(
      <Menuitems listcontent={props.listItem} onClick={handleClick} />
    );
    expect(wrapper).toHaveLength(1);
    expect(handleClick).not.toHaveBeenCalled();

    const spy = jest.spyOn(Menuitems.prototype, 'handleClick');
    for (let index = 0; index < 4; index++) {
      const item = props.listItem[index];
      wrapper
        .find('Link')
        .at(index)
        .simulate('Click', {
          target: {
            isactiveId: item.id,
            location: item.link
          }
        });
      wrapper.setState({
        isactiveId: item.id,
        location: item.link
      });
      expect(spy).toHaveBeenCalled();
      wrapper.update().state();
      console.log(item);
      expect(wrapper.state().isactiveId);
    }
  });
});
