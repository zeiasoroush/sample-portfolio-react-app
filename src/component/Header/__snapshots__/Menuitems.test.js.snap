// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<menuitem /> it render Header 1`] = `ShallowWrapper {}`;

exports[`<menuitem /> specify active tab in header onclick event  1`] = `ShallowWrapper {}`;
